lib/assert.js

```diff
@@ -278,7 +278,13 @@ function expectedException(actual, expected) {
     // Ignore.  The instanceof check doesn't work for arrow functions.
   }
 
-  return expected.call({}, actual) === true;
+  try {
+    return expected.call({}, actual) === true;
+  } catch (e) {
+    // Ignore. If expected is a class constructor, an error will be thrown
+  }
```

Comment:
There are definitely semver implications. I don't have strong feelings about this PR. To be honest, I'd rather close it than special case and rethrow errors.